ui/app/services/path-help.js

```diff
@@ -226,11 +227,16 @@ export default Service.extend({
     const deletePath = paths.find(path => path.operations.includes('delete'));
 
     return generatedItemAdapter.extend({
-      urlForItem(id, isList) {
+      urlForItem(id, isList, dynamicApiPath) {
         const itemType = getPath.path.slice(1);
         let url;
         id = encodePath(id);
-
+        // the apiPath changes when you switch between routes but the apiPath variable does not unless the model is reloaded
+        // check first for the dynamicApiPath which is passed through from the model->adapter
+        // If no dynamicApiPath, then use apiPath set higher up in function
```

Comment:
nit: I find the wording here a bit confusing. It's not exactly checking for dynamicApiPath "first", it's just overwriting `apiPath` if the value exists